packages/pages/src/components/image/image.tsx

```diff
@@ -0,0 +1,210 @@
+    );
+  }
+
+  const imgWidth: number = image.image.width;
+  const imgHeight: number = image.image.height;
+  const imgUUID = getImageUUID(image.image.url);
+
+  const { src, imgStyle, widths } = handleLayout(
+    layout,
+    imgWidth,
+    imgHeight,
+    imgUUID,
+    style,
+    width,
+    height,
+    aspectRatio
+  );
+
+  // Generate Image Sourceset
+  const srcSet: string = widths
+    .map((w) => `${getImageUrl(imgUUID, w, (imgHeight / imgWidth) * w)} ${w}w`)
+    .join(", ");
+
+  const onError = () => {
+    setImgStatus(imgLoadingStatus.ERROR);
+    console.warn(`Invalid image src: ${src}.`);
+  };
+
+  return (
+    <>
+      {placeholder != null &&
+        imgStatus != imgLoadingStatus.LOADED &&
+        placeholder}
+      {imgStatus != imgLoadingStatus.ERROR && (
+        <img
+          ref={imgRef}
+          style={imgStyle}
+          src={src}
+          className={className}
+          width={width}
+          height={height}
+          srcSet={srcSet}
+          loading={"lazy"}
+          onLoad={() => setImgStatus(imgLoadingStatus.LOADED)}
+          onError={() => onError()}
+          {...imgOverrides}
+        />
+      )}
+    </>
+  );
+};
+
+/**
+ * Returns the UUID of an image given its url. Logs a warning message if the image url is invalid.
+ */
+export const getImageUUID = (url: string) => {
+  const matches = url.match(MKTGCDN_URL_REGEX);
+
+  if (matches == null || matches.length == 0) {
+    console.warn(`Invalid image url: ${url}.`);
+    return "";
+  }
+
+  return matches[0];
+};
+
+/**
+ * Returns the image url given its uuid, width and height.
+ */
+export const getImageUrl = (uuid: string, width: number, height: number) => {
+  return `https://dynl.mktgcdn.com/p/${uuid}/${Math.round(width)}x${Math.round(
+    height
+  )}`;
+};
+
+/**
+ * Returns the src, imgStyle and widths that will be set on the underlying img tag based on the
+ * layout.
+ */
+export const handleLayout = (
+  layout: ImageLayout,
+  imgWidth: number,
+  imgHeight: number,
+  imgUUID: string,
+  style: React.CSSProperties,
+  width?: number,
+  height?: number,
+  aspectRatio?: number
+): { src: string; imgStyle: React.CSSProperties; widths: number[] } => {
+  let widths: number[] = [100, 320, 640, 960, 1280, 1920];
+  let src: string = getImageUrl(imgUUID, 500, 500);
+  style.objectFit = style.objectFit || "cover";
+  style.objectPosition = style.objectPosition || "center";
+
+  switch (layout) {
+    case ImageLayout.INTRINSIC:
+      // Don't let image be wider than its intrinsic width
+      style.maxWidth = imgWidth;
+      style.width = "100%";
+      style.aspectRatio = aspectRatio
+        ? `${aspectRatio}`
+        : `${imgWidth} / ${imgHeight}`;
+
+      break;
+    case ImageLayout.FIXED:
+      if (!width && !height) {
+        console.warn(
+          "Using fixed layout but width and height are not passed as props."
+        );
+      }
+
+      const fixedWidth = width
```

Comment:
as a general rule, I try to avoid nested ternary operators, I use tao of react to guide most of my style / design decisions and this is something that's mentioned: https://alexkondov.com/tao-of-react/#avoid-nested-ternary-operators